spras/util.py

```diff
@@ -8,10 +8,22 @@
 from pathlib import Path, PurePath, PurePosixPath
 from typing import Any, Dict, List, Optional, Tuple, Union
 
+import numpy as np
 import pandas as pd
 
 
-def hash_params_sha1_base32(params_dict: Dict[str, Any], length: Optional[int] = None) -> str:
+# https://stackoverflow.com/a/57915246/7589775
+class NpEncoder(json.JSONEncoder):
```

Comment:
Will need to extend this for bool if you accept my other suggestion above.

Reply:
I've accepted the above `bool` suggestion, which was a good catch, but this specifically relates to numpy encoding. I have added a comment that explains this.